### ISSUE TITLE
Discovery of WebRTC connection grant (PING review)

### ISSUE THREAD

**Issue by dontcallmedom** [w3c/webrtc-pc#689]
From the PING review:
> It is troubling that once a WebRTC connection has been granted, could
be very difficult for a non-expert user to discover/revoke -- how is
this concern being addressed?

**Comment by alvestrand** [comment]
Can we have some clarification of what they mean by "a WebRTC connection has been granted"?
I know we grant permission for media sourcing from local devices, but so far, we haven't had the concept of granting permission to WebRTC connections, any more than we grant permission to HTTP connections.

**Comment by alvestrand** [comment]
(sorry, wrong button)

**Comment by gregnorc** [comment]
Hi Alestrand,

I'm part of PING, and participated in the review.

What we meant by a WebRTC connection beting granted is that WebRTC gets access to the user's machine. So just like a user might click on an HTTP link, maybe they click on a webrtc link, or use an application that uses WebRTC. But at the high level, a user in some way initates use of WebRTC, and when this occurs we worry about how the user revokes that permission? (Close the tab? Close the browser? Clear their browser settings? How can they be aware that the action they've taken achieved their goal of revoking WebRTC's access?)

Does this help clarify?

**Comment by alvestrand** [comment]
No, it doesn't make much sense to me. I think we may have a model disconnect.
WebRTC is functionality embedded in the browser. So it runs on the user's machine - it's present when it's installed, and the ability to use WebRTC is implicit in having the browser installed.
There is no such thing as a "webrtc link". There is only Javascript that invokes WebRTC-related functions, in the same way as with all other Javascript-callable functions.
There is no "WebRTC access" that can be granted or revoked. That term simply isn't part of the model.

If you can rephrase what you're afraid of happening in terms that match the illustrations in draft-ietf-rtcweb-overview, I think that might improve the communication.

**Comment by gregnorc** [comment]
Ok, sure. Then let me rephrase:

I am concerned that WebRTC will be activated in the background by a malicious site, and that the user will not be aware this has happened, and have no easy way to revoke consent, other than to close the browser.

Does that make sense?

**Comment by josephlhall** [comment]
@alvestrand We meant the grant for camera and/or mic, not a connection. So this gets back to the question of ease of revoking camera/mic, which I suspect is an issue you've done in the WG to death. I suspect the response would be that the spec specifies that the permission should be revocable but does not get into browser chrome as that's not what w3c does.

I suspect what we would like to see is something in the spec that at least identified that temporary grants or revoking grants shouldn't require expert UA knowledge (like about:* or similar twiddling).

**Comment by dontcallmedom** [comment]
@josephlhall thanks for the clarification; the webrtc spec is "only" about the connection, since the camera/mic permission is managed via [another spec](https://www.w3.org/TR/mediacapture-streams/) the group has been developing in a [separate repo](https://github.com/w3c/mediacapture-main/) and where indeed permission and revokation is discussed at some length (and where we tried to [accommodate PING reviews](https://www.w3.org/2016/03/getusermedia-wide-review.html#ping) as much as possible).

If indeed this issue is about camera/mic permission, I think it ought to be closed. Another one might be open in the other repo if necessary.

**Comment by josephlhall** [comment]
Hard to believe, but I had forgotten about that! That seems like a good plan (closing here and taking anything else up in MediaCapture)

**Comment by alvestrand** [comment]
@gregnorc if by "WebRTC" you mean "camera and microphone access", I finally understand what you're saying! - and fully agree with Dom's statement. Thank you for rephrasing!

**Comment by dontcallmedom** [comment]
Closing per discussion above
